spring-integration-gcp/src/main/java/org/springframework/integration/gcp/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -50,6 +53,11 @@ public PubSubInboundChannelAdapter(PubSubOperations pubSubTemplate,
 			String subscriptionName) {
 		this.pubSubTemplate = pubSubTemplate;
 		this.subscriptionName = subscriptionName;
+
+		StringMessageConverter stringMessageConverter = new StringMessageConverter();
+		stringMessageConverter.setSerializedPayloadClass(String.class);
+		getMessagingTemplate().setMessageConverter(
+				new CompositeMessageConverter(ImmutableList.of(stringMessageConverter)));
```

Comment:
We don't use Guava 😉 

On the other hand your immutability doesn't have effect. See that `CompositeMessageConverter` ctor:
```
public CompositeMessageConverter(Collection<MessageConverter> converters) {
		Assert.notEmpty(converters, "Converters must not be empty");
		this.converters = new ArrayList<>(converters);
	}
```
There is just local copy of your provided list.

OTOH why don't you allow end-user to inject `MessageConverter` at all?

Reply:
I'm not using ImmutableMap so much because of the immutability, but more for syntactic sugar (inline map declaration and construction).
Although I can always declare the map separately.